cost/aws/instance_cloudwatch_utilization/aws_instance_cloudwatch_utilization.pt

```diff
@@ -0,0 +1,666 @@
+  var start_date = new Date(new Date().setDate(new Date().getDate() - 30)).toISOString();
+  if (privateDnsName != null && privateDnsName !== undefined && privateDnsName.length != 0) {
+    var hostname = privateDnsName.split('.')[0]
+  } else {
+    var hostname = "ip-127-0-0-1"
+  }
+  results = {
+    "auth": "monitoring_auth_"+region.replace(/-/g,"_"),
+    "host": 'monitoring.'+region+'.amazonaws.com',
+    "verb": "GET",
+    "path": "/",
+    "headers": {
+      "User-Agent": "RS Policies",
+      "Content-Type": "application/json",
+      "x-amz-target": "GraniteServiceVersion20100801.GetMetricStatistics",
+      "Accept": "application/json",
+      "Content-Encoding": "amz-1.0"
+    }
+    "query_params": {
+      'Action': 'GetMetricStatistics',
+      'Version': '2010-08-01',
+      'Namespace': 'CWAgent',
+      'MetricName': 'disk_used_percent',
+      'Dimensions.member.1.Name': 'host',
+      'Dimensions.member.1.Value': hostname,
+      'Dimensions.member.2.Name': 'device',
+      'Dimensions.member.2.Value': 'xvda1',
+      'Dimensions.member.3.Name': 'fstype',
+      'Dimensions.member.3.Value': 'ext4',
+      'Dimensions.member.4.Name': 'path',
+      'Dimensions.member.4.Value': '/',
+      'StartTime': start_date,
+      'EndTime': end_date,
+      'Period': "2592000",
+      'Statistics.member.1': 'Maximum',
+      'Statistics.member.2': 'Average',
+      'Statistics.member.3': 'Minimum'
+    }
+  }
+EOS
+end
+
+script "js_cloudwatch_mem_usage", type: "javascript" do
+  result "results"
+  parameters "region","instance_id","privateDnsName"
+  code <<-EOS
+  if (privateDnsName != null && privateDnsName !== undefined && privateDnsName.length != 0) {
+    var hostname = privateDnsName.split('.')[0]
+  } else {
+    var hostname = "ip-127-0-0-1"
+  }
+  var end_date = new Date().toISOString()
+  var start_date = new Date(new Date().setDate(new Date().getDate() - 30)).toISOString();
+  results = {
+    "auth": "monitoring_auth_"+region.replace(/-/g,"_"),
+    "host": 'monitoring.'+region+'.amazonaws.com',
+    "verb": "GET",
+    "path": "/",
+    "headers": {
+      "User-Agent": "RS Policies",
+      "Content-Type": "application/json",
+      "x-amz-target": "GraniteServiceVersion20100801.GetMetricStatistics",
+      "Accept": "application/json",
+      "Content-Encoding": "amz-1.0"
+    }
+    "query_params": {
+      'Action': 'GetMetricStatistics',
+      'Version': '2010-08-01',
+      'Namespace': 'CWAgent',
+      'MetricName': 'mem_used_percent',
+      'Dimensions.member.1.Name': 'host',
+      'Dimensions.member.1.Value': hostname,
+      'StartTime': start_date,
+      'EndTime': end_date,
+      'Period': "2592000",
+      'Statistics.member.1': 'Maximum',
+      'Statistics.member.2': 'Average',
+      'Statistics.member.3': 'Minimum'
+    }
+  }
+EOS
+end
+
+script "js_merged_metrics", type: "javascript" do
+  parameters "ds_cloudwatch_cpu_maximum", "ds_cloudwatch_disk_usage", "ds_cloudwatch_mem_usage"
```

Comment:
isn't this `ds_cloudwatch_cpu_maximum` just cpu usage?

Reply:
Yes sir, I updated it to cpu_usage. I probably was trying only maximum at first.